core/hostInfo/cloud_provider.go

```diff
@@ -0,0 +1,44 @@
+package hostInfo
+
+import (
+	"encoding/json"
+	"github.com/rancher/agent/cloudprovider/aws"
+	"github.com/rancher/agent/utilities/config"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+type CloudProviderCollector struct{}
+
+var providers = []string{"aws"}
+
+func (c CloudProviderCollector) GetData() (map[string]interface{}, error) {
+	return nil, nil
+}
+
+func (c CloudProviderCollector) KeyName() string {
+	return "cloudProvider"
+}
+
+func (c CloudProviderCollector) GetLabels(prefix string) (map[string]string, error) {
+	for _, provider := range providers {
```

Comment:
Per my comment in the provider, the collector should just have to read one generic cloud_provider.json file and need not have any knowledge of which provider is in use (if any)